src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +41,87 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+        ProcessHandle processHandle = ProcessHandle.current();
+        ProcessHandle.Info processInfo = processHandle.info();
+        long currentStartTime = processInfo.startInstant().map(Instant::toEpochMilli).orElse(0L);
+
+        List<String> fileContent;
+        if (Files.exists(filePath)) {
+            fileContent = Files.readAllLines(filePath, StandardCharsets.UTF_8);
```

Comment:
```suggestion
        List<String> fileContent = Files.exists(filePath) ? Files.readAllLines(filePath, StandardCharsets.UTF_8) : Collections.emptyList();
```
You can remove the inlining in the whole block